Engines/Wine/Verbs/D9VK/script.js

```diff
@@ -38,22 +37,30 @@ class D9VK {
 
         print("NOTE: Wine version should be greater or equal to 3.10");
 
-        if (operatingSystemFetcher.fetchCurrentOperationSystem() != "Linux") {
-            wizard.message(
-                tr(
-                    "D9VK might not work correctly on macOS. This is depending on Metal api support and MoltenVK compatibility layer advancement"
-                )
-            );
-        } else {
-            wizard.message(
-                tr(
-                    "Please ensure you have the latest drivers (418.30 minimum for NVIDIA and mesa 19 for AMD) or else D9VK might not work correctly."
-                )
-            );
+        if (operatingSystemFetcher.fetchCurrentOperationSystem().getFullName() !== "Linux")
+        {
+            const answer = wizard.menu(
+                tr("D9VK is currently unsupported on non-Linux operating systems due to MoltenVK implementation being incomplete. Select how do you want to approach this situation.",
+                    ["YES, continue with D9VK installation regardless", "NO, quit script alltogether", "Exit D9VK Installer, but continue with the script"]
+                ))
+
+            if (answer.text === "Exit D9VK Installer, but continue with the script") {
+                return this;
+            }
+
+            if (answer.text === "NO, quit script alltogether" || !answer) {
+                throw new Error("User aborted the script.");
+            }
+        }
+        else
+        {
+            wizard().message(tr("Please ensure you have the latest drivers (418.30 minimum for NVIDIA and mesa 19 for AMD) or else D9VK might not work correctly."));
         }
 
-        if (typeof this.d9vkVersion !== "string") {
-            this.d9vkVersion = "0.12";
+        if (typeof d9vkVersion !== 'string')
```

Comment:
`d9vkVersion` is part of the class/object and not a local field inside the `go` method, therefore it must be `this.d9vkVersion`